core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -767,6 +768,20 @@ open class TileGroup(
         }
     }
 
+    private fun ImageAttempter<MapUnit>.tryGetUnitEraSprite(unit: MapUnit, specificUnitIconLocation: String): ImageAttempter<MapUnit> {
+        return this.tryImages(
+            // iterate in reverse order to get the most recent era-specific image
+            (unit.civInfo.getEraNumber() downTo 0).asSequence().map {
+                @Suppress("unused")
+                fun MapUnit.() = if (civInfo.nation.style.isNotEmpty())
```

Comment:
U wot m8
Why do you need to create a fun for this?
Also why are eras by number not by name?

Reply:
> Also why are eras by number not by name?
The logic behind that is so that modders don't have to make so many copies of sprites. If you want the Worker sprite to switch to a guy with a hard hat and a jackhammer starting in the Modern era, you would only have to make a single extra file `Worker-era5.png`. For eras 6, 7, and 8 (Atomic through Future era), they would iterate through the era numbers backwards to find the most recent era where the sprite changed. Going by era name means that the modder would have to copy the identical image for these eras.

Reply:
I understand the "not every era" approach but that would work equally well using the era name, nit the number.
Current era sprites are called by name, and I think the advantage of immediate readability when looking at the sprites (compared to cross referencing with the eras json) vastly outweighs the need to rename sprites of you change an era name.

Because a rename usually indicates that it's supposed to be a different eta, so who knows if it's still relevant?
And what if my mod starts from modern era, I'll need to renumber everything?
I'm not willing to go with numbers.

Reply:
Your identifiers could take precedence over the era name, it alternatively be secondary to it (I think the latter).
But both are good, since they have indicative naming that does not change according to other eras that may exist.

Reply:
https://github.com/yairm210/Unciv/tree/master/android/Images/TileSets/HexaRealm/Tiles

Existing "object-era" naming scheme for tiles

Reply:
Alright, I was able to change the loop to use era names while preserving the "use latest available era image" mechanism.